tinylog-impl/src/main/java/org/tinylog/writers/JsonWriter.java

```diff
@@ -0,0 +1,256 @@
+		}
+		builder.append("{");
+		for (int i = 0; i < tokens.size(); i++) {
+			Token token = tokens.get(i);
+			token.render(logEntry, builder);
+			if (i + 1 < tokens.size()) {
+				builder.append(",");
+			}
+		}
+		builder.append("}");
+		builder.append(",");
+
+		escapeCharacter("\\", "\\\\");
+		escapeCharacter(NEW_LINE, "\\n");
+		escapeCharacter("\t", "\\t");
+		writer.write(builder.toString().getBytes(charset), builder.length());
+	}
+
+	private void escapeCharacter(final String character, final String escapeWith) {
+		int index = builder.indexOf(character);
+		while (index != -1) {
+			builder.replace(index, index + character.length(), escapeWith);
+			index += escapeWith.length();
+			index = builder.indexOf(character, index);
+		}
+	}
+
+	@Override
+	public void flush() throws IOException {
+		writer.flush();
+	}
+
+	@Override
+	public void close() throws IOException {
+		postprocessFile();
+		writer.close();
+	}
+
+	/**
+	 * Preprocesses the JSON file. If append mode is on, deletes the closing bracket
+	 * and adds a comma instead. If it's a new file, appends an opening bracket.
+	 * 
+	 * @param append Append Mode on or off
+	 * @throws IOException              Error reading or writing file
+	 * @throws IllegalArgumentException Invalid file format
+	 */
+	private void preprocessFile(final boolean append) throws IOException, IllegalArgumentException {
+		if (append && inputChannel.size() > 0) {
+			long sizeToTruncate = 0;
+			long currentPosition = inputChannel.size();
+			boolean foundClosingBracket = false;
+			while (!foundClosingBracket) {
+				long from = Math.max(0, currentPosition - BUFFER_SIZE);
+				long numberOfBytes = Math.min(currentPosition, BUFFER_SIZE);
+				MappedByteBuffer section = inputChannel.map(FileChannel.MapMode.READ_ONLY, from, numberOfBytes);
+				byte[] bytes = new byte[section.remaining()];
+				section.get(bytes);
+				if (bytes.length == 0) {
+					throw new IllegalArgumentException(
+							"Invalid JSON file. The file is missing a closing bracket for the array.");
+				}
+				for (int i = bytes.length - 1; i >= 0; i--, currentPosition--) {
+					sizeToTruncate += 1;
+					if (bytes[i] == ']') {
+						foundClosingBracket = true;
+						break;
+					}
+				}
+			}
+			long newFileSize = fileChannel.size() - sizeToTruncate;
+			fileChannel.truncate(newFileSize);
+			writer.write(commaByte, 1);
+		}
+
+		if (inputChannel.size() == 0) {
+			writer.write(bracketOpenByte, 1);
+		}
+	}
+
+	/**
+	 * Postprocesses the JSON file. Attempts to delete the trailing comma and
+	 * appends closing bracket which were handled in
+	 * {@link #preprocessFile(boolean)}
+	 * 
+	 * @throws IOException Error writing to file
+	 */
+	private void postprocessFile() throws IOException {
+		fileChannel.truncate(fileChannel.size() - commaByte.length);
+		writer.write(NEW_LINE.getBytes(charset), 1);
+		writer.write(bracketCloseByte, 1);
+	}
+
+	@Override
+	public Collection<LogEntryValue> getRequiredLogEntryValues() {
```

Comment:
For me, it would be easier to find methods if they are ordered my visibility. Firstly, all public methods, and all private methods at the end of the class.

Reply:
Fixed in https://github.com/tinylog-org/tinylog/pull/195/commits/784dbde0077f730872e7550ea33778c142f7293f